sycl/source/detail/usm/usm_impl.cpp

```diff
@@ -232,6 +232,11 @@ void *alignedAllocInternal(size_t Alignment, size_t Size,
 void *alignedAlloc(size_t Alignment, size_t Size, const context &Ctxt,
                    const device &Dev, alloc Kind, const property_list &PropList,
                    const detail::code_location &CodeLoc) {
+  if (Dev.has(sycl::aspect::usm_device_allocations) == false) {
```

Comment:
Depending on `Kind`, we should check either `usm_device_allocations` or `usm_shared_allocations`. I doubt there are relevant devices with one but not the other, but the spec says what the spec says.

Reply:
@al42and Thanks for that!
All your remarks I just applied in newest commit.
Please take a look